File: app/src/main/java/bin/mg/main/PullToRefreshLayout.java
package bin.mg.main;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.widget.FrameLayout;

public class PullToRefreshLayout extends FrameLayout {

    public interface PullToRefreshListener {
        void onRefresh(PullToRefreshLayout pullToRefreshLayout);
    }

    private PullToRefreshListener listener;

    public PullToRefreshLayout(Context context) {
        this(context, null);
    }

    public PullToRefreshLayout(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public PullToRefreshLayout(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void setPullToRefreshListener(PullToRefreshListener listener) {
        this.listener = listener;
    }

    public void setRefreshing(boolean refreshing) {
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        return false;
    }

    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        return true;
    }
}

File: app/src/main/java/bin/mg/main/PullToRefreshLayout.kt
package bin.mg.main

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.widget.FrameLayout

class PullToRefreshLayout @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : FrameLayout(context, attrs, defStyleAttr) {

    interface PullToRefreshListener {
        fun onRefresh(pullToRefreshLayout: PullToRefreshLayout)
    }

    private var listener: PullToRefreshListener? = null

    fun setPullToRefreshListener(listener: PullToRefreshListener?) {
        this.listener = listener
    }

    fun setRefreshing(refreshing: Boolean) {
        // Stub
    }

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        return false
    }

    override fun onTouchEvent(ev: MotionEvent): Boolean {
        return true
    }
}
